src/Layouts/Partials/TransformPanel.vala

```diff
@@ -205,17 +218,54 @@ public class Akira.Layouts.Partials.TransformPanel : Gtk.Grid {
 
         attach (group_title (_("Opacity")), 0, 9, 3);
         attach (opacity_grid, 0, 10, 3);
+
+        event_bus.selected_items_changed.connect (on_selected_items_changed);
+    }
+
+    private void on_selected_items_changed (List<Lib.Models.CanvasItem> selected_items) {
+        if (selected_items.length () == 1) {
+            item = selected_items.nth_data (0);
+        } else {
+            if (item != null) {
+                item = null;
+            }
+        }
```

Comment:
This looks pretty dirty. Let's try to always avoid nested conditions and handle the negatives first.
eg:

```
if (!selected_items.length ()) {
  item = null;
  return;
}

item = selected_items.nth_data (0);
```

Reply:
Ok, totally feasible!